### PR TITLE
Mccalluc/workspaces join sections

### LINKED ISSUE #2785: Use workspaces API to put jobs under workspaces
Right now they are two separate sections: Make use of https://github.com/hubmapconsortium/user_workspaces_server/issues/53 to join them.

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #2785

It seemed best to me to bury the logic for merging the two API responses inside one hook, but you may have thoughts about it.

### CODE REVIEW

**Comment by mccalluc** [comment]
(Moving back to draft... let me add a little unit test.)

File: context/app/static/js/components/workspaces/utils.js [hunk 1]
```diff
@@ -41,4 +41,22 @@ async function startJob({ workspaceId, workspacesEndpoint, workspacesToken }) {
...
+function mergeJobsIntoWorkspaces(jobs, workspaces) {
+  const wsIdToJobs = {};
+  jobs.forEach((job) => {
+    const { workspace_id } = job;
+    if (!(workspace_id in wsIdToJobs)) {
+      wsIdToJobs[workspace_id] = [];
+    }
+    wsIdToJobs[workspace_id].push(job);
+  });
+
```
Comment (john-conroy) [inline context/app/static/js/components/workspaces/utils.js:53]:
```suggestion
  jobs.reduce((acc, job) => {
      const { workspace_id } = job;
      if(!(workspace_id in acc)){
          acc[workspace_id] = [];
      }
      acc[workspace_id] = [...acc[workspace_id], job];
      return acc;
      }, {});

```

Comment (john-conroy) [reply context/app/static/js/components/workspaces/utils.js:53]: A slight preference for reduce, but up to you. The indentation on the suggestion seems slightly off.

**Review by john-conroy — APPROVED** [review]

File: context/app/static/js/components/workspaces/utils.js (hunk 1 above)
Comment (mccalluc) [reply context/app/static/js/components/workspaces/utils.js:53]: I'd need to do something with the result of the `reduce`, right? It's not my usual idiom, but I'm wondering if there are arguments for it beyond:
- No external scratch variable.
- Favoring functional style over procedural.

Also, wondering about the array construction... the point is to favor creating new objects over mutating existing? I worry that it hurts readability.

Comment (mccalluc) [reply context/app/static/js/components/workspaces/utils.js:53]: (Going to fix indentation and merge as is, but happy to learn more about your motivation.)

Comment (mccalluc) [reply context/app/static/js/components/workspaces/utils.js:53]: ... and I realize I don't see the problem with the indentation, either. Sorry not to get it.

**Pull request merged**